counsel.el

```diff
@@ -5306,10 +5306,12 @@ to further refine the compile options in the directory specified by `blddir'."
 (defun counsel-compile (&optional dir)
   "Call `compile' completing with smart suggestions, optionally for DIR."
   (interactive)
-  (add-hook 'compilation-start-hook 'counsel-compile--update-history)
+  ;; No need to specify `:history' because of this hook.
+  (add-hook 'compilation-start-hook #'counsel-compile--update-history)
```

Comment:
@abo-abo did mention moving this to somewhere else but I'm not sure I see anywhere else being particularly better or worse.

Reply:
The idea is that we modify a public hook without removing ourselves from it, and we don't even make use of the argument passed to each function in the abnormal hook. This isn't the end of the world, but it indicates that there's probably a cleaner way to call `counsel-compile--update-history`. I'm not yet familiar enough with this new logic to make any suggestions as to where this place could be.